### PR TITLE
"migrate" to cats effect 3 

### PR DESCRIPTION
**Pull request by b3nk3i**
removed dependency to io.monix which is not yet available for CE3..

### CODE REVIEW

**Comment by kubukoz** [comment]
@LukaJCB

**Comment by kubukoz** [comment]
What was the monix dependency for? The build seems fine without it.

**Comment by TimWSpence** [comment]
If we could get this released I would be immensely grateful :) Let me know if I can help in any way

**Comment by b3nk3i** [comment]
> What was the monix dependency for? The build seems fine without it.

it was used as example in the doc : policies.md

**Comment by kubukoz** [comment]
ah, I see it now.

**Review by LukaJCB — APPROVED** [review]
LGTM

File: modules/core/shared/src/main/scala/retry/Sleep.scala [hunk 1]
```diff
@@ -10,6 +11,6 @@ trait Sleep[M[_]] {
 object Sleep {
   def apply[M[_]](implicit sleep: Sleep[M]): Sleep[M] = sleep
 
-  implicit def sleepUsingTimer[F[_]](implicit timer: Timer[F]): Sleep[F] =
-    (delay: FiniteDuration) => timer.sleep(delay)
+  implicit def sleepUsingTemporal[F[_]](implicit t: Temporal[F]): Sleep[F] =
```
[the remark this answers is not in the document]
Comment (b3nk3i) [reply modules/core/shared/src/main/scala/retry/Sleep.scala:14]: From what I see, replacing Sleep by Temporal has a bigger impact: 
given that Temporal[F[_]] = GenTemporal[F, Throwable] 

You end up with MonadError[F, Throwable] following the hierarchy GenTemporal[F, E] ... GenConcurrent[F, E] ... GenSpawn[F, E] ... MonadCancel[F, E] ... MonadError[F, E]

Then Temporal[F] will clash with internals based on MonadError[F, E], though it seems that GenTemporal[F, E] solves it but the change propagates up to public function.. not sure if that's desired.. I think it's probably better to try it in a separate PR

[the remark this answers is not in the document]
Comment (kubukoz) [reply modules/core/shared/src/main/scala/retry/Sleep.scala:14]: Fair enough, these are good points :)

**Review by kubukoz — APPROVED** [review]
LGTM

File: build.sbt [hunk 2]
```diff
@@ -56,9 +56,9 @@ val moduleSettings = commonSettings ++ Seq(
   scalafmtOnCompile := true
 )
 
-val catsVersion          = "2.3.0"
-val catsEffectVersion    = "2.3.1"
-val catsMtlVersion       = "1.1.1"
+val catsVersion          = "2.5.0"
+val catsEffectVersion    = "3.0.2"
```
[the remark this answers is not in the document]
Comment (b3nk3i) [reply build.sbt]: version updated

File: build.sbt [hunk 3]
```diff
@@ -56,9 +56,9 @@ val moduleSettings = commonSettings ++ Seq(
...
+val catsMtlVersion       = "1.1.3"
```
[the remark this answers is not in the document]
Comment (b3nk3i) [reply build.sbt]: version updated

File: build.sbt [hunk 4]
```diff
@@ -134,7 +134,7 @@ val docs = project
       "org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full
     ),
     libraryDependencies ++= Seq(
-      "io.monix" %%% "monix" % "3.1.0"
+      //"io.monix" %%% "monix" % "3.3.0" // TODO update to version compatible cats effect 3
```
[the remark this answers is not in the document]
Comment (b3nk3i) [reply build.sbt]: dropped

**Comment by LukaJCB** [comment]
Thanks a bunch!

**Comment by kubukoz** [comment]
Can we have a milestone for this?

**Comment by b3nk3i** [comment]
> Can we have a milestone for this?

The build is failing on a test based on Future.. could be a sporadic failure as it passed on the PR (also locally), can we retry the build ? If it fails again maybe increasing that test timeout will help

**Comment by cb372** [comment]
I'm late to the party but LGTM 😄 Thanks for the PR.

Thanks to @LukaJCB this has just been released as v3.0.0.

**Pull request merged**